server/src/main/java/org/elasticsearch/script/JodaCompatibleZonedDateTime.java

```diff
@@ -63,19 +64,20 @@
     private static final DateFormatter DATE_FORMATTER = DateFormatter.forPattern("strict_date_time");
     private static final DeprecationLogger deprecationLogger = DeprecationLogger.getLogger(JodaCompatibleZonedDateTime.class);
 
-    private static void logDeprecated(String key, String message, Object... params) {
+    private static void logDeprecated(DeprecationCategory category, String key, String message, Object... params) {
         AccessController.doPrivileged(new PrivilegedAction<Void>() {
             @SuppressLoggerChecks(reason = "safely delegates to logger")
             @Override
             public Void run() {
-                deprecationLogger.deprecate(key, message, params);
+                deprecationLogger.deprecate(category, key, message, params);
```

Comment:
Consider just passing in `DeprecationCategory.PARSING` here instead of altering the methods below. Since this method is private I think it's safe to say it'll be used for parsing issues within this class.